x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -910,6 +910,7 @@ Collection<Object> createComponents(
         final AuthorizationService authzService = new AuthorizationService(
             settings,
             allRolesStore,
+            fieldPermissionsCache,
```

Comment:
Do we want to make this change? The cache used by RBACEngine is techinically cross different index groups and the combination depends on the requested indices patterns, e.g. if we have 
```
"indices": [
  { 
    "name": ["logs-*"], "privileges": ["read"], 
    "field_security": {"grant": ["xxx"] }  
  },
  { 
    "name": ["*-staging"], "privileges": ["read"], 
    "field_security": {"grant": ["yyy"] }  
  },
]
```

If the requested index pattern is `logs-nginx-staging`, both groups will match and the cache key will be 
```
new FieldPermissionsDefinition(
  Set.of(
    new FieldGrantExcludeGroup(["xxx"], null), 
    new FieldGrantExcludeGroup(["yyy"], null)
  )
)
```

While the cache used in CompositeRolesStore is most likely keyed by a single index group. With the above example, the two groups will be cached separately. If we combine the two caches, it is possible that more entries would be competing for cache usages. 

Along the same line of reasoning, the cache used by RoleDescriptor is purely for a single index definition. Strictly speaking, it can also be different from the cache usage by CompositeRolesStore which can combine index groups if they have exactly the same `names` definition. 

In summary, the cache is keyed slightly different with the 3 use cases:
1. RoleDescriptor - keyed strictly by a single index group's FLS definition
2. CompositeRolesStore - keyed by FLS combination of multiple index groups that share the same `names` definition
3. RBACEngine - keyed by FLS combination of mulitple index groups that have `names` matching the requested index. 

That said, I do think having a single cache for them is reasonable because we don't really want to differentiate clearly between the different type of cache keys. My only concern is whether combining them could add extra noticeable pressure on this single cache.